### PR TITLE
add Popover docs

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Changes to docs: added new `/popover` page with usage and accessibility details.

While working on docs, I found I had to make changes to Popover component (now moved into #1602).

## Testing

Tested all examples in multiple browsers and with keyboard.

## Docs

n/a. These *are* docs.

### CODE REVIEW

File: examples/Popover.focus.tsx
Comment (mayank99) [inline examples/Popover.focus.tsx:24]: I had to add `border: none` override because Surface always insists on bringing its own border.

Also, for some reason `Surface.Body` only brings horizontal padding not vertical.


@FlyersPh9 Is this intentional?

File: packages/itwinui-react/src/core/Popover/Popover.tsx
Comment (gretanausedaite) [inline packages/itwinui-react/src/core/Popover/Popover.tsx:107]: 1. I liked that we are giving the users this feature. I think it would be useful for them.
2. If we want to make it private, prop imports in components that use Popover needs to be updated as it is using only public props.

Comment (mayank99) [reply packages/itwinui-react/src/core/Popover/Popover.tsx:107]: i can't think of any use case where the user might want `matchWidth`, it's mostly useful for menus (which we already provide multiple components for). do you have another example in mind? if a user asks for it, we can always add it back in the future.

i've fixed the props types in [`d146a85`](https://github.com/iTwin/iTwinUI/pull/1598/commits/d146a859002a0e53538c30395b813a2ef02eee71)

**Review by mayank99 — COMMENTED** [review]
@rohan-kadkol Reading through these docs will help you make sense of the changes in #1602

**Review by gretanausedaite — APPROVED** [review]

File: examples/Popover.focus.tsx
Comment (rohan-kadkol) [inline examples/Popover.focus.tsx:40]: nit: It might be helpful if there were an explanation comment here. Something like, "To grab focus when the popover opens".

**Review by rohan-kadkol — APPROVED** [review]

File: examples/Popover.focus.tsx
Comment (mayank99) [inline examples/Popover.focus.tsx:33]: @gretanausedaite @rohan-kadkol @siddhantrawal Can somebody suggest better text or a different example? I dunno how this has 3 approvals with TODO text 😅

Comment (rohan-kadkol) [reply examples/Popover.focus.tsx:33]: > Can somebody suggest better text or a different example?

How about a "see image preview" popover? If we click on the trigger button that says "show image preview", the popover shows the image preview along with an "OK" button and a "Download" button.

> I dunno how this has 3 approvals with TODO text 😅

I thought that TODO was for you so you will handle it yourself 😅 So I gave a preemptive review to not block the PR because everything else LGTM.

File: examples/Popover.focus.tsx
Comment (FlyersPh9) [reply examples/Popover.focus.tsx:24]: `data-iui-padded="true"` does have vertical padding, it's just very little.
Screenshot 2023-09-28 at 4 28 04 PM
The examples on [the surface demo page](https://itwin.github.io/iTwinUI/css/surface.html) look nice, but the agree button in your screenshot does not look so great.  Was the padding changed when we implemented `scrollbar-gutter`?

Btw, just now found out [`scrollbar-gutter` doesn't work in Safari](https://caniuse.com/mdn-css_properties_scrollbar-gutter). 😓

File: examples/Popover.focus.tsx
Comment (gretanausedaite) [reply examples/Popover.focus.tsx:33]: > Can somebody suggest better text or a different example?

Some sliders to "tune settings", eg. quality, grain, color, saturation, etc. (like in photoshop) with ok or save / cancel buttons at the bottom.

Comment (gretanausedaite) [reply examples/Popover.focus.tsx:33]: > I dunno how this has 3 approvals with TODO text

We have more than one todo in repo that is supposed to be solved later.

Comment (mayank99) [reply examples/Popover.focus.tsx]: updated the example to show a settings dialog

Comment (gretanausedaite) [reply examples/Popover.focus.tsx]: It's good. We have this in css fieldset tests if you want to switch inputs but you don't have to. This example is to show focus control, it does not have to be useful or realistic.
image

**Review by rohan-kadkol — APPROVED** [review]

**Pull request merged**
